ovs-p4rt/sidecar/spies/ovsp4rt_config_fdb_entry_spy.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Test double (spy) for the ovsp4rt_config_fdb_entry API.
+ * Dumps the input parameters in JSON.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+#include "ovsp4rt/ovs-p4rt.h"
+
+void ovsp4rt_config_fdb_entry(struct mac_learning_info learn_info,
```

Comment:
I read the notes and description on the main page of PR.

I still dont think I follow what the intention is - are we planning a refactor of functions such as `ovsp4rt_config_fdb_entry`?

Reply:
I'm planning to do two things:

1. Focus on creating approval tests, rather than unit tests, for OVSP4RT.
2. Take another shot at refactoring OVSP4RT, using the approval tests to verify that I haven't altered the behavior of the code.

To do this, I need to:

1. Create a set of functions that serialize the inputs to each of the public APIs.
2. Instrument OVSP4RT to capture the inputs and outputs of each of the public API functions.
3. Create a set of test cases in OVS. Use the instrumentation to capture the corresponding transactions in OVSP4RT.
4. Create a suite of approval tests that can feed the inputs into OVSP4RT, capture the outputs, and check for deviations from the expected results.

I started with `ovsp4rt_config_fdb_entry()` because that's the API for which I wrote the unit test. I used it to learn how to serialize a set of inputs into a json object. The next step is to instrument its input and output paths, and then rejigger the unit test to create a corresponding approval test.

I suspect the `ovsp4rt_spies` library will morph into a support library for the instrumented version of OVSP4RT as well as the approval tests. And I've been wondering whether I might be able to use it in conjunction with some of the programs in the OVS test suite...

Reply:
I am ok with creating approval tests and have no issues with that.

With regards to refactoring, please check with @vsureshkumarp and @n-sandeep on the repurcussions. Suresh was suspecting that the last round of refactoring may have messed with the timing mechanism and the flow (since each individual method itself seemed ok), which caused the errors/issues.

Reply:
Approving this PR, since there are no functional changes

Reply:
> With regards to refactoring, please check with @vsureshkumarp and @n-sandeep on the repercussions. Suresh was suspecting that the last round of refactoring may have messed with the timing mechanism and the flow (since each individual method itself seemed ok), which caused the errors/issues.

That seems unlikely. The last round of refactoring is gone.